### PR TITLE
Node ID-based navigation for document list

### PR DESCRIPTION
**Pull request by DenisVuyka**
refs #1366 

Addresses the following issues:
- DocumentList fails navigating user home subfolders #1366 
- Get the correct node path for a normal user #1280
- Document List should deal with node ID rather than path #505
- Custom root support for Breadcrumb #528

Greatly reduced Breadcrumb code, no more relative path parsing (`node.path` exposed by `alfresco-js-api` is used instead)

**Introduces breaking changes**:

- `currentFolderPath` property is no longer supported and replaced with a combination of `currentFolderId: string` and `folderNode: MinimalNodeEntryEntity` properties

### CODE REVIEW

File: ng2-components/ng2-alfresco-documentlist/README.md [hunk 1]
```diff
@@ -162,17 +161,13 @@ export class AppModule {
...
 ```
 
 ### Properties
 
 | Name | Type | Default | Description |
 | --- | --- | --- | --- |
-| `rootFolderId` | string | -root- | Root node ID, i.e. `-root-`, `-shared-`, `-my-`, etc. or a fixed node ID |
-| `currentFolderPath` | string | null | Initial path of displayed folder below the root node, e.g. "/Sites/swsdp/documentLibrary" |
-| `currentFolderId` | string | null | Initial node ID of displayed folder, if given  |
+| `currentFolderId` | string | null | Initial node ID of displayed folder. Can be `-root-`, `-shared-`, `-my-`, or a fixed node ID  |
```
Comment (wabson) [inline ng2-components/ng2-alfresco-documentlist/README.md]: Does `folderNode` input need describing here too?

File: ng2-components/ng2-alfresco-documentlist/src/components/document-list.spec.ts [hunk 2]
```diff
@@ -158,69 +150,46 @@ describe('DocumentList', () => {
...
-        spyOn(documentList, 'loadFolderByPath').and.returnValue(Promise.resolve());
-        documentList.currentFolderPath = null;
-        documentList.ngOnChanges({currentFolderPath: new SimpleChange('', null)});
-
-        expect(documentList.currentFolderPath).toBe(documentList.DEFAULT_FOLDER_PATH);
-        expect(documentList.loadFolderByPath).toHaveBeenCalledWith(documentList.DEFAULT_FOLDER_PATH);
+        expect(documentList.loadFolder).not.toHaveBeenCalled();
     });
 
+    /*
```
Comment (wabson) [inline ng2-components/ng2-alfresco-documentlist/src/components/document-list.spec.ts]: Remove commented out test or replace `it()` with `xit()` if there is a reason to keep it for now?

File: ng2-components/ng2-alfresco-documentlist/README.md (hunk 1 above)
Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-documentlist/README.md]: Yes, I will extend readme

**Comment by DenisVuyka** [comment]
@wabson updated the code as per discussion

**Pull request merged**